src/platform_impl/android/mod.rs

```diff
@@ -938,16 +947,51 @@ impl Window {
         ))
     }
 
-    pub fn raw_window_handle(&self) -> RawWindowHandle {
+    #[cfg(feature = "rwh_04")]
+    pub fn raw_window_handle_rwh_04(&self) -> rwh_04::RawWindowHandle {
+        if let Some(native_window) = self.app.native_window().as_ref() {
+            let mut handle = rwh_04::AndroidNdkHandle::empty();
+            handle.a_native_window = native_window.ptr().as_ptr() as *mut _;
+            rwh_04::RawWindowHandle::AndroidNdk(handle)
+        } else {
+            panic!("Cannot get the native window, it's null and will always be null before Event::Resumed and after Event::Suspended. Make sure you only call this function between those events.");
+        }
+    }
+
+    #[cfg(feature = "rwh_05")]
+    pub fn raw_window_handle_rwh_05(&self) -> rwh_05::RawWindowHandle {
+        use rwh_05::HasRawWindowHandle;
+
         if let Some(native_window) = self.app.native_window().as_ref() {
             native_window.raw_window_handle()
         } else {
             panic!("Cannot get the native window, it's null and will always be null before Event::Resumed and after Event::Suspended. Make sure you only call this function between those events.");
         }
     }
 
-    pub fn raw_display_handle(&self) -> RawDisplayHandle {
-        RawDisplayHandle::Android(AndroidDisplayHandle::empty())
+    #[cfg(feature = "rwh_05")]
+    pub fn raw_display_handle_rwh_05(&self) -> rwh_05::RawDisplayHandle {
+        rwh_05::RawDisplayHandle::Android(rwh_05::AndroidDisplayHandle::empty())
+    }
+
+    #[cfg(feature = "rwh_06")]
+    pub fn raw_window_handle_rwh_06(&self) -> Result<rwh_06::RawWindowHandle, rwh_06::HandleError> {
```

Comment:
Question while implementing the `ndk` portion for this: implementing the `Has**RAW**WindowHandle` trait is deprecated and there are blanket implementations when `HasWindowHandle` is implemented. Should this function also return a non-raw handle (in terms of that implementation when it is released)?

Reply:
I considered this, but it would add an additional `unsafe { WIndowHandle::borrow_raw }` call for each backend. I could go either way.

Reply:
Right. At least for Android I'm willing to have that `unsafe { borrow_raw() }` inside the `ndk` with appropriate `// SAFETY` docs, so you could call through to that. But it'd have to be consistent across all the backends...

Reply:
Did you consider whether or not the safe borrow handles should be implemented by `winit`?

Reply:
For the time being it leads to a lot more `unsafe` throughout the codebase. So I think that we shouldn't for now.